macros/src/lib.rs

```diff
@@ -58,7 +58,13 @@ impl DerivedTS {
         };
 
         let export = match self.export {
-            true => Some(self.generate_export_test(&rust_ty, &generics)),
+            true => {
+                if cfg!(feature = "test-export") {
```

Comment:
The following would be more elegant here:
```rust
let export = match self.export {
    #[cfg(feature = "test-export")]
    true => { .. }
    _ => None
```
This would also work:
```rust
```rust
let export = match self.export {
    true if cfg!(feature = "test-export") => { .. }
    _ => None
```

```